### ISSUE TITLE
Deploying to the web breaks the application

### ISSUE THREAD

**Issue by smalers** [OpenWaterFoundation/owf-app-infomapper-ng#209]
Trying to deploy the latest  0.9.1 InfoMapper to the web breaks the Poudre Basin Information website.  The error is related to `L` not being defined.  I updated the [`copy-to-owf-amazon-s3.sh`](https://github.com/OpenWaterFoundation/owf-infomapper-poudre/blob/master/build-util/copy-to-owf-amazon-s3.sh) script to have a `--nooptimize` option so that the files don't get minimized.  The command line is as follows.  The `--optimization=false` argument turns off minification.

```
ng build --prod=true --aot=true --baseHref=. --extractCss=true --namedChunks=false --outputHashing=all --sourceMap=false --optimization=false
```

I also had to update the budget limits to 20MB but this should not be the value in the repo.  The InfoMapper should use 7MB for the error limit on budgets.  Josh needs to run something similar for the InfoMapper test dataset and then try to run the application in the `dist` folder.  This should point out the same issues as the AWS-deployed version.  Doing this requires getting the root folder right so paths work.  The broken version is:  [Poudre 0.2.1](http://poudre.openwaterfoundation.org/0.2.1/).

**Comment by Nightsphere** [comment]
I have made a new push for this issue. I changed the way georaster-layer-for-leaflet is imported into the app. Just like last
time, it works both in ng serve and with ng build. Hopefully this will help.

**Comment by smalers** [comment]
Below is a screen shot of the error:

![image](https://user-images.githubusercontent.com/6448274/88349713-1263fb80-cd0e-11ea-8b0d-207178ad5e6b.png)

**Comment by smalers** [comment]
I tried to read up on webpack and found [this article](https://blog.jakoblind.no/how-webpack-decides-what-to-bundle/).

So, I did:  `find . -name webpack.config.js` and it listed: `./node_modules/georaster/webpack.config.js`.  However, there were no other instances of this file found.  Maybe this is in conflict with what Angular normally does.  I also found [this article](https://developer.okta.com/blog/2019/12/09/angular-webpack).

**Comment by smalers** [comment]
I wan to help Josh be able to troubleshoot.  I hope that the following works even without all the data.  If not, I can help provide some additional files.

I added the [`build-util/run-dist-server-8000.sh`](https://github.com/OpenWaterFoundation/owf-infomapper-poudre/blob/master/build-util/run-dist-server-8000.sh) script to the Poudre Basin Information app.

Test the Poudre Basin Information application as follows:

1. Setup the development environment by cloning/pulling it and running `build-util/git-clone-all-prod.sh`.  Then make sure the InfoMapper is set up as usual.
2. Run `build-util/copy-to-owf-amazon-s3.sh --noupload --nooptimization` to build the `dist` version but don't upload.  This version won't minify the code, which make it possible to see the original code.
3. Run `build-util/run-dist-server-8000.sh` and then open the application with `http://local:8000`.  The error in the previous comment is shown.  Hopefully Josh can duplicate this to allow for troubleshooting.

**Comment by Nightsphere** [comment]
After a few hours looking into the issue, I wouldn't say I know exactly what's going on behind the scenes, but I was able to fix the issue in a few different ways. I have listed them in order from lowest to highest potential impact of the app in the future, so they should be attempted from 1 to 3.

1. Everything stays the same as before with the exception of the main-es2015 script tag. `defer` needs to be added after the `type="text/javascript"` as follows:
```javascript
<script src="main-es2015.3c2070bf8d0b1491e484.js" type="text/javascript" defer>
```
If this is a stable fix, we can probably update `text/javascript` to `application/javascript` in the future.

2. Add `defer` to *every* script tag in `index.html` that doesn't already have it. This might be redundant, as 1. seems to be working just fine, but this also didn't give me any issues.

3. The `main-es2015` file can be left alone, so that Angular causes the script tag to create normally, resulting in
```javascript
<script src="main-es2015.3c2070bf8d0b1491e484.js" type="module">
```
This contains defer by default, and will also take care of the issue. The `runtime-es2015` and `polyfills-es2015` files would still
need to have `type="text/javascript"`. Also, this may still cause the AWS issue that plagued us before, so it is number 3 on this list. I will add to this issue if any other/better solutions present themselves or if this isn't a stable fix.
